### PR TITLE
Support `self-update` command for self version updating

### PR DESCRIPTION
**Pull request by moznion**
This pull request introduces a new subcommand `self-update`. This subcommand updates the running cli binary itself if it finds the newer version CLI.

Usage example:

```
$ soracom version
SORACOM API client v0.0.0
$ soracom query sims --name XXXXXXXXX
The latest version v0.11.0 is released which is newer than current version v0.0.0. Please update to the latest version. 'soracom self-update' command is handy to update it.
The detailed release note of the latest version is here: https://github.com/soracom/soracom-cli/releases/tag/v0.11.0
[]
$ soracom self-update
There is a newer soracom-cli version 'v0.11.0'.
Release Note: https://github.com/soracom/soracom-cli/releases/tag/v0.11.0
Do you want to continue updating? (y/N) y
It updated successfully.
$ soracom version
SORACOM API client v0.11.0
```

### CODE REVIEW

**Comment by bearmini** [comment]
Thank you for the great enhancement!

One point I want to clarify is whether this approach also works for `soracom` binary installed with `brew`.
In my environment, the result of `which soracom` returns `/home/linuxbrew/.linuxbrew/bin/soracom` but it is a symlink to `/home/linuxbrew/.linuxbrew/bin/Cellar/soracom-cli/0.11.0/bin/soracom`. 

According to godoc, `os.Executable()` returns:
> If a symlink was used to start the process, depending on the operating system, the result might be the symlink or the path it pointed to. 

In either case, there is a concern that simply replacing the binary may destroy the internal state of brew.

What do you think?

**Comment by bearmini** [comment]
We also provide an installation method with `snap`. This may have the same problem.

**Comment by moznion** [comment]
Thank you for the comment, @bearmini.
Right, I had not thought about the symlink issue that you mentioned. I made a patch to make it track the path to the actual binary if the running one is a symlink @ [f5024ef](https://github.com/soracom/soracom-cli/pull/51/commits/f5024ef861cda0a2d4139b607e53db256d654b9b). What do you think about this?

And in the case of the `self-update` trails the actual path, I simulated what will it occur if the CLI-tool attempts `self-update` and replace the executable binary that is managed by homebrew.

```
$ readlink /usr/local/bin/soracom
../Cellar/soracom-cli/0.10.4/bin/soracom
$ soracom version
SORACOM API client v0.10.4
# ↓ replace the actual binary with v0.11.0 manually (i.e. simulation of `self-update`)
$ sudo curl -L -o /usr/local/Cellar/soracom-cli/0.10.4/bin/soracom https://github.com/soracom/soracom-cli/releases/download/v0.11.0/soracom_0.11.0_darwin_amd64
$ soracom version
SORACOM API client v0.11.0
$ brew update
$ brew upgrade soracom-cli
==> Upgrading 1 outdated package:
soracom/soracom-cli/soracom-cli 0.10.4 -> 0.11.0
==> Downloading https://github.com/soracom/soracom-cli/releases/download/v0.11.0/soracom_0.11.0_darwin_amd64
Already downloaded: /Users/moznion/Library/Caches/Homebrew/downloads/26b0ddebf196f24654406ca508591e55445f563d423a90cb4a313c611d44b92d--soracom_0.11.0_darwin_amd64
==> Upgrading soracom/soracom-cli/soracom-cli
  0.10.4 -> 0.11.0

🍺  /usr/local/Cellar/soracom-cli/0.11.0: 3 files, 12.2MB, built in 3 seconds
$ readlink /usr/local/bin/soracom
../Cellar/soracom-cli/0.11.0/bin/soracom
$ soracom version
SORACOM API client v0.11.0
```

It seems updated successfully, so I assume the actual binary and that metadata are managed separately. So even if the binary is managed by something package managers, thus I suppose it's okay to update by `self-update`; of course, it would be better to use the proper way to update according to the managers.

File: generators/assets/cli/en.yaml [hunk 1]
```diff
@@ -255,3 +255,9 @@ cli:
     description: List, create, update or delete VPGs.
   deprecated-api: Deprecated API calling.
   alternative-api-suggestion: "Please consider to use the alternative API (command: `%s`)."
+  self_update:
+    summary: Updates soracom-cli to the latest version.
+    description: Updates soracom-cli to the latest version.
+    already_latest: "It doesn't need to update the CLI because the using version is the latest: %s\n"
+    prompt_confirmation: "There is a newer soracom-cli version '%[1]s'.\nRelease Note: https://github.com/soracom/soracom-cli/releases/tag/%[1]s\nDo you want to continue updating? (y/N) "
```
Comment (bearmini) [inline generators/assets/cli/en.yaml:262]: How about adding some sentences to encourage the users to use appropriate package managers to update the binary, like:

```
There is a newer soracom-cli version '$VERSION'.
Release Note: http://github.com/soracom/soracom-cli/releases/tag/$VERSION

If you installed soracom-cli using a package manager, please use it to update.

  brew update soracom-cli
    or
  snap refresh soracom
    or
  dpkg -i soracom_$VERSION_$ARCH.deb

Do you wan to continue updating? (y/N) 
```

Comment (moznion) [reply generators/assets/cli/en.yaml]: Awesome, that seems helpful. I added that notes @ [8974235](https://github.com/soracom/soracom-cli/pull/51/commits/8974235d97c6f89245fdc89c9eff8f5a42f4180e)

**Review by bearmini — APPROVED** [review]
Awesome! Thank you for the great contribution!

**Pull request merged**